### PR TITLE
DOC: Add examples for pandas.tseries.timedeltas

### PR DESCRIPTION
**Pull request by proinsias**
Copied examples from the reStructuredText [tutorial](http://pandas.pydata.org/pandas-docs/stable/timedeltas.html) on timedeltas.

### CODE REVIEW

File: pandas/tseries/timedeltas.py [hunk 1]
```diff
@@ -32,6 +32,20 @@ def to_timedelta(arg, unit='ns', box=True, errors='raise', coerce=None):
     Returns
     -------
     ret : timedelta64/arrays of timedelta64 if parsing succeeded
+
+    Examples
+        --------
```
Comment (jorisvandenbossche) [inline pandas/tseries/timedeltas.py]: You did something strange with the indentation here

File: pandas/tseries/timedeltas.py [hunk 2]
```diff
@@ -32,6 +32,20 @@ def to_timedelta(arg, unit='ns', box=True, errors='raise', coerce=None):
...
+
+        >>> pd.to_timedelta('1 days 06:05:01.00003')
+        Timedelta('1 days 06:05:01.000030')
+        >>> pd.to_timedelta('15.5us')
+        Timedelta('0 days 00:00:00.000015')
+        >>> pd.to_timedelta(['1 days 06:05:01.00003','15.5us','nan'])
+        TimedeltaIndex(['1 days 06:05:01.000030', '0 days 00:00:00.000015', NaT], dtype='timedelta64[ns]', freq=None)
+        >>> pd.to_timedelta(np.arange(5),unit='s')
```
Comment (jorisvandenbossche) [inline pandas/tseries/timedeltas.py]: can you add a space after the comma (PEP8)

**Comment by jorisvandenbossche** [comment]
@proinsias Thanks a lot for working on examples in docstrings!

Besides the inline comments, another general: maybe it would be nice to have some text between the different examples to somehow put them a bit into context, eg:

```
Parsing a single string to a Timedelta:

>>> ...
>>> ...

Parsing a list or array of strings:

>>> ...

Converting numbers by specifying the `unit` keyword argument:

>>> ...

```

**Comment by proinsias** [comment]
Thanks for the advice! – I've made these changes and re-pushed the branch.

File: pandas/tseries/timedeltas.py [hunk 3]
```diff
@@ -32,6 +32,25 @@ def to_timedelta(arg, unit='ns', box=True, errors='raise', coerce=None):
     Returns
     -------
     ret : timedelta64/arrays of timedelta64 if parsing succeeded
+
+    Examples
+    --------
+
+    Parsing a single string to a Timedelta:
+    >>> pd.to_timedelta('1 days 06:05:01.00003')
```
Comment (jorisvandenbossche) [inline pandas/tseries/timedeltas.py]: Small formatting note: there needs to be a blank line between the two lines above (between the code blocks and the text, so the code blocks are rendered as code, see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections for a full reference)

**Comment by jreback** [comment]
thanks!

**Pull request merged**